server/src/Lib.hs

```diff
@@ -91,12 +109,36 @@ server (ServerConfig programDB appDB _) =
     :<|> (\bin -> handle bin Nothing)
     :<|> (\pkg bin -> handle bin (Just pkg))
   where
-    handle :: BinaryName -> Maybe PackageName -> Maybe Platform -> Handler ByteString
-    handle bin mpkg msys =
+    handle :: BinaryName -> Maybe PackageName -> Maybe Platform -> Maybe NixpkgsCommit -> Handler ByteString
+    handle bin mpkg msys mcommit =
       Handler . withExceptT (\err -> err400 {errBody = BSL.pack err}) $ do
+        commit <- resolveCommit mcommit
         let sys = fromMaybe X86_64_Linux msys
         pkg <- maybe (resolvePackageName programDB bin sys) pure mpkg
-        buildTriplet appDB (BinaryTriplet bin pkg sys)
+        buildTriplet appDB (BinaryTriplet bin pkg sys) commit
+
+-- | The default Nixpkgs commit we use when the user doesn't provide one.
+--
+-- This is nixos-22.05 as of 2022-08-29.
+defaultNixpkgsCommit :: NixpkgsCommit
+defaultNixpkgsCommit = NixpkgsCommit "0ba2543f8c855d7be8e90ef6c8dc89c1617e8a08"
```

Comment:
I think it's a good idea to specify the default commit as a command line argument. That way, we can use nix to pass `inputs.nixpkgs.rev` as the default commit, and automatically keep it in sync with the flake. What do you think?

Reply:
I agree that it would be nice to take the commit we're using as be a command line argument (although maybe you'll let me do this in a follow-up PR?).

But I don't think we should keep this commit in sync with our flake.  My justification is that there really isn't relation between the Nixpkgs commit we use to build our server, and the Nixpkgs commit that we use for serving these statically-linked exes.  

In the future we probably want the flexibility to, for instance, build our server with `nixos-unstable` with ghc-9.2.4, while still using 22.05 to serve statically-linked exes.

Further in the future I imagine we also might allow building from multiple Nixpkgs commits (for instance both 22.05 and 22.11).

Reply:
> But I don't think we should keep this commit in sync with our flake

I was thinking the same thing. It's up to you whether you want to add the command line argument already, I'd be fine with doing it in a separate PR.